lib/rules/no-unused-prop-types.js

```diff
@@ -1041,10 +1041,19 @@ module.exports = {
       },
 
       JSXSpreadAttribute: function(node) {
-        const component = components.get(utils.getParentComponent());
-        components.set(component ? component.node : node, {
-          ignorePropsValidation: true
-        });
+        // includes `{...props}`, `{...others}` etc.
+        const isIdentifier = node.argument.type === 'Identifier';
```

Comment:
Does this mean that if i do `<div {...someRandomThing} />` it will cause props not to be marked as unused, as well?

Reply:
Yes, it's not good. But I found a test case use `{...other}` should props be marked as used.  
This test case and `<div {... someRandomThing} />` seems have conflict.

https://github.com/yannickcr/eslint-plugin-react/blob/8237551309e2beaaa751d3d3fcc8bdb15e0f9a1b/tests/lib/rules/no-unused-prop-types.js#L756-L763

Reply:
That should be marked as used because `other` resolves to the props object. If it *doesn't* resolve to that, it should not be marked as used.

Reply:
I debug into the method `JSXSpreadAttribute`.  
Both the 2 test case as below the parameter `node` object is the same.  
It seems difficult to verify the `node` object does resolve to the props object or not.
Could you give me some suggestions? Thanks. @ljharb 

```js
// props should be marked as used
render() {
  var {...other} = this.props;
  render (<div {...other} />); // `node` object is `{...other}`

// props should be marked as unused
render() {
  const other = [{ className: 'foo' }];
  render (<div {...other} />);  // `node` object is `{...other}` too
```

Reply:
That's a good question, and I'm not sure. You may need to set up visitors for a bunch of other node types, and keep track of their names and what name they point to?

Reply:
My thought is: 
1. find all nodes relate with the jsx spread attribute name (which node name is equal with the jsx spread attribute name, such as `other`) in the component.
2. check every related node whether resolves to the props object or not.
3. if yes, set the `ignorePropsValidation` to true and break loop.

But I'm not familiar with AST. I think it's difficult for me to handle this problem 😂.  
Maybe I need someone help.

Reply:
Perhaps we could make the rule a bit looser.

The case as below should be marked as used, others are not: 
* `{...this.props}`
* `{...props}` (not matter what `props` is)

Reply:
that's probably a better place to leave it then wrongly marking them as used :-)